### PR TITLE
Data iterator cleanup 11 29 2023

### PR DESCRIPTION
**Pull request by caterpillarAttack**
Alrighty,

This is a big cleanup of various for loops and logic:
- Conversion to higher-order functions where applicable, I.E. map, filter, any, next. (This should speed things up, or at the least make stuff much more legible and the intentions of the programmer clearer.
- Conversion to list comprehensions to increase speed.
- Added docstring stubs for type hint checking in the IDE.
- Removed some unused variables.
- Conversion to using .extend() for list concatenations rather than operator overloading. This makes it clearer what is going on and marginally speeds stuff up.
- Reduced the overall LOC of the data file by a couple percentile.

### CODE REVIEW

File: python/vtool/data.py [hunk 1]
```diff
@@ -1390,11 +1398,8 @@ def import_skin_weights(self, directory, mesh, first=True):  # TODO: This beast
 
         if util_file.is_file(file_path):
             lines = util_file.get_file_lines(file_path)
-
-            for line in lines:
-                test_line = line.strip()
-                if not test_line:
-                    continue
+            for line in filter(lambda x: bool(x),
```
Comment (caterpillarAttack) [inline python/vtool/data.py:1401]: use a len() here instead of the bool.

File: python/vtool/data.py [hunk 2]
```diff
@@ -4046,40 +3974,26 @@ def export_data(self, comment, selection=None):  # TODO: Refactor
...
                     selected_node_names = controller.get_graph().get_select_nodes()
-
-                    for node_name in selected_node_names:
-                        node_inst = controller.get_graph().find_node(node_name)
-                        if node_inst:
-                            found.append(node_inst)
-
-                if found:
-                    nodes += found
+                    found = [node_inst for node_inst in
```
Comment (caterpillarAttack) [inline python/vtool/data.py:3981]: This should be changed, we are both overwriting and extending the declaration here.

File: python/vtool/data.py [hunk 3]
```diff
@@ -1021,35 +1034,16 @@ def _get_influences(self, folder_path):
...
-            if filename == 'all.skin.weights':
-                found_single_file_weights = True
-                continue
-
-            if not filename.endswith('.weights'):
-                continue
-
-            influences.append(filename)
+        found_single_file_weights = any(filter(lambda x: x == 'all.skin.weights', files))
+        influences = [filename for filename in files if filename.endswith('.weights')]
```
Comment (caterpillarAttack) [inline python/vtool/data.py:1038]: Careful, it has a continue for the original iterator, add an and to the comprehension.

File: python/vtool/data.py [hunk 4]
```diff
@@ -646,11 +665,8 @@ def import_data(self, filename=None, selection=None):
             selection = []
         library = self._initialize_library(filename)
 
-        controls = []
         if selection:
-            for thing in selection:
-                maya_lib.core.has_shape_of_type(thing, 'nurbsCurve')
```
Comment (caterpillarAttack) [inline python/vtool/data.py:652]: Was this lookup intentional or intended to be part of an if statement originally? @louisVottero

File: python/vtool/data.py [hunk 5]
```diff
@@ -678,11 +694,8 @@ def export_data(self, comment, selection=None):
             selection = []
         library = self._initialize_library()
 
-        controls = []
         if selection:
-            for thing in selection:
-                maya_lib.core.has_shape_of_type(thing, 'nurbsCurve')
```
Comment (caterpillarAttack) [inline python/vtool/data.py:684]: Was this lookup intentional or intended to be part of an if statement originally? @louisVottero

File: python/vtool/data.py (hunk 3 above)
Comment (caterpillarAttack) [reply python/vtool/data.py]: resolved.

File: python/vtool/data.py (hunk 4 above)
Comment (caterpillarAttack) [reply python/vtool/data.py:652]: Marking as resolved, as current code changes handle the check.

**Review by louisVottero — APPROVED** [review]

**Pull request merged**